resources/js/data-access/memo-api/groups/System.ts

```diff
@@ -0,0 +1,23 @@
+import { V1 } from "../config";
+import { FacilityResource } from "../resources";
+import { Api } from "../types";
+import { parseGetListResponse } from "../utils";
+
+export namespace System {
+  export const getFacilitiesList = () =>
+    V1.get<Api.Response.GetList<FacilityResource>>(
+      "/system/facility/list"
+    ).then(parseGetListResponse);
+
+  export const keys = {
+    all: () => ["system"] as const,
+    facilityAll: () => [...keys.all(), "facility"] as const,
+    facilityLists: () => [...keys.facilityAll(), "list"] as const,
+    facilityList: () => [...keys.facilityLists()] as const,
+  };
+
+  export const facilitiesQuery = {
```

Comment:
Jak już wspominałem, nie jestem przekonany do tego. Niby stała, niby nie wiadomo co w niej mogłoby się zepsuć od używania jej, ale jednak robimy defensywną kopię? Moim zdaniem albo musi być na to dobry powód, albo trzeba to przerobić na coś bardziej oczywistego, np. funkcję zwracającą gotowe query. Zwłaszcza tutaj, gdzie nie ma naprawdę co parametryzować jeszcze w tym query, nie spodziewamy się chyba że ktoś będzie dodawał do tych dwóch jakąś masę innych parametrów.

A już _co najmniej_ to mi wygląda na facilitiesQueryOptions, a nie facilitiesQuery.

Reply:
ok, zgodzę się, że można to nazwać jako options. W takich obiektach chciałbym tworzyć podstawową konfigurację danego query, aby potem każdy subsrciber mógł to dowolnie nadpisać. Najważniejsze są queryFn oraz querKey, bo one są niezmienne w kontekście jednego query. Zrezygnowałem z custom hooków do query, bo potem trzeba typować te funkcje i mogą być mniej elastyczne niż takie podejście. Aczkolwiek oba podejścia są prawie tożsame